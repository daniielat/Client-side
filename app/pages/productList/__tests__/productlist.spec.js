const React = require('react');
const ProdListView = require('../view');
const { render, screen, act, fireEvent } = require('@testing-library/react');
// require('core-js');
// const restClient = require('nordic/restclient');
// jest.mock('nordic/restclient');

// const { mockGet } = restClient;

describe('La view de Home', () => {
    let component; 

    beforeEach(async() => {  
        await act(() =>{

            component= render(<ProdListView />);
        })
    });
    
    it('1) Renderiza', () => {
        const { asFragment } = component;
        expect(asFragment()).toMatchSnapshot();
    });

    it('2) Si no hay productos deberia mostrar el msj "no se encontraron productos"', () => {
        const text = screen.getByText(/no se encontraron productos/i)
        expect(text).toBeInTheDocument();
    })
        
    xit('3) Debe mostrar la lista de productos al hacer click en el boton', async() => {
        await act(async () => {
            const button = await screen.findByRole('pagination');
            fireEvent.click(button);
        })

        const item = await screen.findByRole('listitem')
        screen.debug(item)
    })
})