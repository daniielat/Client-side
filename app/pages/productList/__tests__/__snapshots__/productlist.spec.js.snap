// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`La view de Home 1) Renderiza 1`] = `
<DocumentFragment>
  <div
    class="demo"
  >
    <h1>
      productList
    </h1>
    <button
      role="pagination"
    >
      NEXT
    </button>
    <ol>
      <h4>
        No se encontraron productos
      </h4>
    </ol>
  </div>
</DocumentFragment>
`;
